h/api/test/models_test.py

```diff
@@ -65,120 +66,486 @@ def groups(pattern, text):
     return re.search(pattern, text).groups() or []
 
 
+def test_uri():
+    assert models.Annotation(uri="http://foo.com").uri == "http://foo.com"
+
+
+def test_uri_with_no_uri():
+    assert models.Annotation().uri == ""
+
+
+def test_uri_is_escaped():
+    spam_link = '<a href="http://example.com/rubies">Buy rubies!!!</a>'
+    annotation = models.Annotation(uri='http://</a>' + spam_link)
+
+    uri = annotation.uri
+
+    assert jinja2.escape(spam_link) in uri
+    for char in ['<', '>', '"', "'"]:
+        assert char not in uri
+
+
+def test_uri_returns_Markup():
+    assert isinstance(
+        models.Annotation(uri="http://foo.com").uri, jinja2.Markup)
+
+
+def test_uri_when_uri_is_not_a_string():
+    for uri in (True, None, 23, 23.7, {"foo": False}, [1, 2, 3]):
+        assert isinstance(models.Annotation(uri=uri).uri, unicode)
+
+
+@patch("h.api.models.Annotation.uri", new_callable=PropertyMock)
```

Comment:
As I mentioned the other day, there's really no need to use `PropertyMock` here. `@patch.object(models.Annotation, 'uri', 'http://example.com/example.html')` would work just fine and would be less to understand.